java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -56,23 +55,26 @@ public void run(File source, File basePath) {
                 resources.add((CustomResourceDefinition) deserialized);
             }
 
-            for (HasMetadata resource : resources) {
-                if (resource.getKind()
-                        .toLowerCase(Locale.ROOT)
-                        .equals("customresourcedefinition")) {
-                    CustomResourceDefinition crd = (CustomResourceDefinition) resource;
-
-                    List<WritableCRCompilationUnit> writables =
-                            generate(crd, getPackage(crd.getSpec().getGroup()));
-
-                    for (WritableCRCompilationUnit w : writables) {
-                        w.writeAllJavaClasses(basePath);
-                    }
-                } else {
-                    LOGGER.warn(
-                            "Not generating nothing for resource of kind: " + resource.getKind());
-                }
-            }
+            resources.parallelStream()
```

Comment:
This is the simplest approach to divide the work. However it will use the number of available cores in your machine to spawn that many threads.
We should probably explore other options (in the future) such as a queue and a configurable executor service. This jobs are IO intensive, so it's very likely that the performance can be improved even more.

Reply:
Yup, this was extremely simple to get in and already provided great value, but I'm open(and eager) to look for alternatives 😄 .

I'm not 100% sure about the "IO intensive" part, a lot of time is spent in the `generateJava` function which is pure in-memory computation (e.g. spawning javaparser `CompilationUnit` etc. etc.), but we will measure next time we iterate over performance.

Reply:
Talking about CI pipelines, AFAIK we are using [default GH runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) and they have 2 cores each, so I would expect an improvement in performance even there (although it is still not the "target machine").

Reply:
I wasn't talking about ours, but about what users might have in place (especially those running on Kubernetes).
Anyway, I will probably experiment with this to get a better feeling.